fvm/executionParameters.go

```diff
@@ -34,7 +36,23 @@ func getEnvironmentMeterParameters(
 
 	sth.DisableAllLimitEnforcements()
 
-	env := NewScriptEnvironment(context.Background(), ctx, vm, sth, programs)
+	txnProgs, err := blockProgs.NewSnapshotReadTransactionPrograms(
+		snapshotTime,
+		executionTime)
+	if err != nil {
+		return params, err
+	}
+
+	defer func() {
+		commitErr := txnProgs.Commit()
```

Comment:
q: do we also need the `ShouldCommitPrograms()` check here?

Reply:
I've added ShouldCommitPrograms to deal with bootstrapping.  I'll rename `ShouldCommitPrograms` to `IsBootstrapping` to make it less confusing.